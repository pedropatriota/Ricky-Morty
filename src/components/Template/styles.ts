import styled from 'styled-components';

export const Container = styled.div`
	width: 100%;
	height: fit-content;
	padding: 0.5rem min(5vw, 5rem);

	div {
		cursor: pointer;
	}
`;
