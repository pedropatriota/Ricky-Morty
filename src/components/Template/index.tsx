import { ReactNode } from 'react';
import { ArrowBigLeft } from 'lucide-react';
import { Container } from './styles';

interface ITemplateProps {
	children: ReactNode;
	showGoBack?: boolean;
	handleGoBack?: () => void;
}

const Template = ({ children, showGoBack, handleGoBack }: ITemplateProps) => {
	return (
		<Container data-testid="template-container">
			{showGoBack && (
				<div>
					<ArrowBigLeft
						onClick={handleGoBack}
						width="30px"
						role="button"
						aria-label="goback-icon"
					/>
				</div>
			)}
			{children}
		</Container>
	);
};

export default Template;
