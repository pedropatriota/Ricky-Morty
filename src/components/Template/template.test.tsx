import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import Template from '.';

describe('Template', () => {
	const handleGoBack = jest.fn();

	it('should render correctly', () => {
		const { rerender } = render(
			<Template showGoBack handleGoBack={handleGoBack}>
				<div>test</div>
			</Template>
		);

		const icon = screen.getByRole('button', { name: 'goback-icon' });
		expect(icon).toBeTruthy();

		const container = screen.getByTestId('template-container');
		expect(container).toHaveStyle('padding: 0.5rem min(5vw, 5rem)');

		rerender(
			<Template showGoBack={false}>
				<div>test</div>
			</Template>
		);

		expect(icon).not.toBeInTheDocument();
	});

	it('should back to Home page when the icon is clicked', async () => {
		render(
			<Template showGoBack handleGoBack={handleGoBack}>
				<div>test</div>
			</Template>
		);

		const icon = screen.getByRole('button', { name: 'goback-icon' });

		fireEvent.click(icon);

		await waitFor(() => {
			expect(global.window.location?.href).toContain(`/`);
		});
	});
	it('should match the snapshot', () => {
		const { container } = render(
			<Template showGoBack handleGoBack={handleGoBack}>
				<div>test</div>
			</Template>
		);
		expect(container).toMatchSnapshot();
	});
});
