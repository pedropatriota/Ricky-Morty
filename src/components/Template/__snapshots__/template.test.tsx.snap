// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template should match the snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg ynpvd"
    data-testid="template-container"
  >
    <div>
      <svg
        aria-label="goback-icon"
        class="lucide lucide-arrow-big-left"
        fill="none"
        height="24"
        role="button"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="30px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 15h-6v4l-7-7 7-7v4h6v6z"
        />
      </svg>
    </div>
    <div>
      test
    </div>
  </div>
</div>
`;
